envisage/plugin_manager.py

```diff
@@ -177,15 +178,17 @@ def stop_plugin(self, plugin=None, plugin_id=None):
     # The plugins that the manager manages!
     _plugins = List(IPlugin)
 
-    def __plugins_changed(self, trait_name, old, new):
+    @observe("_plugins")
+    def _update_application_on_plugins(self, event):
         """ Static trait change handler. """
-
+        old, new = event.old, event.new
         self._update_plugin_application(old, new)
 
-    def __plugins_items_changed(self, trait_name, old, new):
+    @observe("_plugins:items")
+    def _update_application_on_changed_plugins(self, event):
```

Comment:
nit: I find these names a little confusing as the one above should update the application on a change to `_plugins`, but not its contents.

TBH though I don't have a better alternative in mind that would make sense.  Feel free to ignore

Reply:
i renamed them to what i think is better.